crates/events/src/enclave_event/compute_request/trbfv.rs

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+//
+// This file is provided WITHOUT ANY WARRANTY;
+// without even the implied warranty of MERCHANTABILITY
+// or FITNESS FOR A PARTICULAR PURPOSE.
+
+use serde::{Deserialize, Serialize};
+
+/// TrBFV modules defining the API for multithreaded compute
+/// Each module defines the event payloads that make up a compute request
+/// Each compute request should live independently and be self contained
+
+/// Input format for TrBFVRequest
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub enum TrBFVRequest {
+    GenEsiSss(gen_esi_sss::Request),
+    GenPkShareAndSkSss(gen_pk_share_and_sk_sss::Request),
+    GenDecryptionKey(gen_decryption_key::Request),
+    GenDecryptionShare(gen_decryption_share::Request),
+    ThresholdDecrypt(threshold_decrypt::Request),
+}
+
+/// Result format for TrBFVResponse
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub enum TrBFVResponse {
+    GenEsiSss(gen_esi_sss::Response),
+    GenPkShareAndSkSss(gen_pk_share_and_sk_sss::Response),
+    GenDecryptionKey(gen_decryption_key::Response),
+    GenDecryptionShare(gen_decryption_share::Response),
+    ThresholdDecrypt(threshold_decrypt::Response),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub enum TrBFVError {
+    // Add errors here as required
+}
+
+pub mod gen_esi_sss {
+    /// This module defines event payloads that will generate esi smudging noise shamir shares to be shared with other members of the committee.
+    /// This has been separated from the general setup in order to be able to take advantage of parallelism
+    use crate::bytes::Bytes;
+    use e3_crypto::SensitiveBytes;
+    use serde::{Deserialize, Serialize};
+
+    #[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+    pub struct Request {
+        /// BFV Params
+        params: Bytes,
+        /// Max number of ciphertexts
+        max_num_ciphertexts: u64,
+    }
```

Comment:
How are we passing the party information to the actor? It needs to know `num_parties` & `threshold` as well for `trbfv` and `ShareManager`.